crates/matrix-sdk-indexeddb/src/crypto_store/mod.rs

```diff
@@ -534,7 +537,17 @@ impl IndexeddbCryptoStore {
 
         let mut device_store = indexeddb_changes.get(keys::DEVICES);
 
+        let account_info = self.get_static_account();
+        let mut clear_caches = false;
         for device in device_changes.new.iter().chain(&device_changes.changed) {
+            // If our own device key changes, we need to clear the session
+            // cache because the sessions contain a copy of our device key, and
+            // we want the sessions to use the new version.
+            if account_info.clone().is_some_and(|info| {
```

Comment:
Do we need the `clone` here? I'd expect to be able to just inspect the value so could do it by reference.

Reply:
We need the `clone` because `is_some_and` takes over the `account_info`, and we're in a `for` loop. Though, taking another look at it, replacing `clone` with `as_ref` seems to make the borrow checker happy, and looks like a better thing to do.